api-service/database/licenses_test.go

```diff
@@ -0,0 +1,338 @@
+			map[string]interface{}{"_id": "Real Application Clusters", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0}}
+
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+}
+
+func (m *MongodbSuite) TestListLicenses() {
+	defer m.db.Client.Database(m.dbname).Collection("hosts").DeleteMany(context.TODO(), bson.M{})
+	m.InsertHostData(utils.LoadFixtureMongoHostDataMap(m.T(), "../../fixture/test_apiservice_mongohostdata_17.json"))
+
+	defer m.db.Client.Database(m.dbname).Collection("licenses").DeleteMany(context.TODO(), bson.M{})
+	m.InsertLicense(model.LicenseCount{Name: "Oracle ENT"})
+	m.InsertLicense(model.LicenseCount{Name: "Diagnostics Pack"})
+	m.InsertLicense(model.LicenseCount{Name: "Real Application Clusters"})
+
+	m.T().Run("should_not_filter", func(t *testing.T) {
+		sortBy := ""
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := ""
+		environment := ""
+		olderThan := utils.MAX_TIME
+		out, err := m.db.ListLicenses(sortBy, sortDesc, page, pageSize, location, environment, olderThan)
+		m.Require().NoError(err)
+
+		var expectedOut []interface{} = []interface{}{
+			//foobar3
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "Oracle EXE", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "Oracle ENT", "purchasedLicense": 0.5},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "Oracle STD", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "WebLogic Server Management Pack Enterprise Edition", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar3", "licenseName": "Diagnostics Pack", "purchasedLicense": 0.5},
+			//foobar4
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "Oracle EXE", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "Oracle ENT", "purchasedLicense": 0.5},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "Oracle STD", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "WebLogic Server Management Pack Enterprise Edition", "purchasedLicense": 0},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "Diagnostics Pack", "purchasedLicense": 0.5},
+			map[string]interface{}{"hostname": "test-db3", "dbName": "foobar4", "licenseName": "Real Application Clusters", "purchasedLicense": 1.5},
+		}
+
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+
+	m.T().Run("should_sort", func(t *testing.T) {
+		sortBy := "licenseName"
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := ""
+		environment := ""
+		olderThan := utils.MAX_TIME
```

Comment:
The parameters take a significant amount of the lines. Why do you not have putted directly into the function parentesis?

Reply:
Removed!